torchtext/experimental/vectors.py

```diff
@@ -0,0 +1,211 @@
+                    csvWriter.writerow(line.split(' ', 1))
+            f2.close()
+        f1.close()
+
+    file_obj = open(csv_file_path, 'r')
+    vectors_obj = vectors_from_file_object(file_obj)
+    save_vectors(vectors_obj, vectors_file_name)
+
+    return vectors_obj
+
+
+def glo_ve(name="840B", dim=300, unk_tensor=None):
+    r"""Create a GloVe Vectors object.
+
+    Args:
+        name (str): the language to use for FastText.
+        unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token.
+
+    Returns:
+        Vectors: a Vectors object.
+
+    """
+    urls = {
+        '42B': 'http://nlp.stanford.edu/data/glove.42B.300d.zip',
+        '840B': 'http://nlp.stanford.edu/data/glove.840B.300d.zip',
+        'twitter.27B': 'http://nlp.stanford.edu/data/glove.twitter.27B.zip',
+        '6B': 'http://nlp.stanford.edu/data/glove.6B.zip',
+    }
+
+    url = urls[name]
+
+    vectors_file_name = os.path.join('.data', 'glove.{}.{}d.pt'.format(name, str(dim)))
+    saved_vectors = load_vectors(vectors_file_name)
+    if saved_vectors:
+        return saved_vectors
+
+    downloaded_file_path = download_from_url(url)
+    extracted_file_path = extract_archive(downloaded_file_path)[0]
+
+    csv_file_name, _ext = os.path.splitext(extracted_file_path)
+    csv_file_path = '{}.csv'.format(csv_file_name)
+
+    # skip if csv file already exists
+    if not os.path.exists(csv_file_path):
+        # format downloaded file into csv format expected by `vectors_from_file_object()`
+        with io.open(extracted_file_path, encoding="utf8") as f1:
+            with open(csv_file_path, "w") as f2:
+                csvWriter = csv.writer(f2)
+                cnt = 0
+                for line in f1:
+                    csvWriter.writerow(line.split(' ', 1))
+            f2.close()
+        f1.close()
+
+    file_obj = open(csv_file_path, 'r')
+    vectors_obj = vectors_from_file_object(file_obj)
+    save_vectors(vectors_obj, vectors_file_name)
+
+    return vectors_obj
+
+
+def vectors_from_file_object(file_like_object, unk_tensor=None):
+    r"""Create a Vectors object from a csv file like object.
+
+    Note that the tensor corresponding to each vector is of type `torch.float`.
+
+    Format for csv file:
+        token1,num1 num2 num3
+        token2,num4 num5 num6
+        ...
+        token_n,num_m num_j num_k
+
+    Args:
+        file_like_object (FileObject): a file like object to read data from.
+        unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token.
+
+    Returns:
+        Vectors: a Vectors object.
+
+    """
+    readCSV = csv.reader(file_like_object, delimiter=',')
+
+    tokens = []
+    vectors = []
+    for row in readCSV:
+        tokens.append(row[0])
+        vectors.append(torch.tensor([float(c) for c in row[1].split()], dtype=torch.float))
+
+    return Vectors(tokens, vectors, unk_tensor=unk_tensor)
+
+
+def save_vectors(vector_obj, path_pt):
```

Comment:
What's the difference with `torch.save`? Why not use it here?

Reply:
`torch.save()` basically saves a pickled model to disc. are you suggesting to use the `torch.save()` function directly in my method?

Reply:
I mean `torch.save(vector_obj, "vector.pt")` should already work here, right? So we don't expect users to use `save_vectors` func.

Reply:
Oh I meant to make this a private function. I was thinking I might be doing a few other checks when loading files to make sure the file exists. I can just call `torch.save` and `torch.load` directly though because it doesn't seem like it warrants its own private function.